### PR TITLE
Cache tox environments on Windows and Mac OS X

### PR DESCRIPTION
**Pull request by sk1p**
Adds parameters that allows recreating the tox environment, and/or run slow tests:

image

This is meant for more comprehensive testing before cutting a release. Some additional tests that took over 10 seconds on any of the platforms are now also marked as slow.

This also disables some of the Azure specific code coverage reporting, as that is sometimes really slow. In general, it appears that some of the jobs are just scheduled on systems that are I/O overloaded - even checking out the source code, unpacking the pip cache etc. takes a long time sometimes:

image

image

image

Refs #1045

## TODO

- [ ] Check coverage when manually running slow tests - maybe merge w/ coverage of previous `tox` run

## Contributor Checklist:

* [ ] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [ ] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [ ] I have added/updated documentation for all user-facing changes
* [ ] I have added/updated test cases
* [ ] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)
## Reviewer Checklist:

* [ ] `/azp run libertem.libertem-data` passed

### CODE REVIEW

**Comment by sk1p** [comment]
@uellue @twentyse7en some of your tests have been disabled for CI in this PR. Could you please review these, and see if there may be another way to test these parts of the code?

In total, we are losing about 1% coverage with this PR (the currently displayed number may not be accurate, as I just triggered a manual run including the slow tests).

**Comment by sk1p** [comment]
This is a run including the slow tests: https://dev.azure.com/LiberTEM/LiberTEM/_build/results?buildId=843&view=results

The coverage looks off a bit, as it should now be back to "normal". It's possible that the coverage for `slow`-marked tests needs to be merged manually, I'll have a look into this later.

**Comment by uellue** [comment]
@sk1p Looks good! Did we see improvements already? I'll have a look at the tests that you marked as slow. Should I push them into this PR?

**Comment by sk1p** [comment]
> Did we see improvements already?

Yes - the slow tests take about 2 to 3 minutes, if you check the link above. With slow tests disabled, we are in total maybe 5 minutes faster, if things go right, so we still have quite some room for improvement, maybe in the long tail...

**Comment by uellue** [comment]
When evaluating the test times we should keep in mind that some tests use session-scoped fixtures that are quite heavy to create and may also require significant warm-up when they are used for the first time -- namely the local Dask executor. The same goes for tests that use Numba-compiled code. Some test will always draw the short straw and be a lot slower than the others. The clustering template happens to be run first from all the template tests, so that contributes to it appearing very slow.

**Comment by sk1p** [comment]
> Some test will always draw the short straw and be a lot slower than the others. 

Yes, indeed - that's probably also the reason `test_com_with_zero_frames` is so slow. This goes into the direction of #1041 - if the numba compilation load is reduced, first-use times should improve a lot.

> The clustering template happens to be run first from all the template tests, so that contributes to it appearing very slow.

I'm not convinced that's the whole story, as the "normal" clustering analysis test was also really slow. The `template/test_pick_analysis` test is much, much faster than, let's say the SD or clustering notebook tests...

**Comment by uellue** [comment]
> I'm not convinced that's the whole story

Yes, it contributes only a part of the extra time. Discussion with @sk1p: It seems to be the import of sklearn. A candidate to move to a separate module, at some point.

**Comment by sk1p** [comment]
Coverage merging via `--cov-append` seems to work (according to `--cov-report=term`), we are just hitting a bug in codecov.io it seems, similar to [this one](https://community.codecov.com/t/codecov-uploads-from-azure-pipelines-are-failing-with-build-numbers-do-not-match/1100). I'm not too worried about that, I think it's a glitch in codecov.
